extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -448,17 +448,31 @@ export class SubscriptionService {
    * @return {!Promise<boolean>}
    */
   delegateActionToService(action, serviceId) {
-    // TODO: add a promise to wait for resolve promise of a platform.
-    const platform = dev().assert(this.platformStore_.getPlatform(serviceId),
-        'Platform is not registered');
-    this.subscriptionAnalytics_.event(
-        SubscriptionAnalyticsEvents.ACTION_DELEGATED,
-        {
-          action,
-          serviceId,
-        }
-    );
-    return platform.executeAction(action);
+    return this.platformStore_.whenPlatformResolves(serviceId)
```

Comment:
Consider the redirect case due to promise->window.open. One option is to put body in the private method and (a) call it immediately if platform already resolved. Or, (b) call it once platform is resolved. Another idea would be to redefine onPlatformResolves with a callback - callbacks are by nature synchronous more often than async. E.g.:

```
this.paltformStore_.onPlatformResolved(serviceId, platform => {
  platform.executeAction(action);
})
```

This way, the Platform will either execute the action _synchronously_ if available, or it will wait and execute async.

Reply:
Done